src/main/java/frc/team2412/robot/commands/intake/IntakeBitmapCommand.java

```diff
@@ -69,39 +81,40 @@ public IntakeBitmapCommand(IntakeSubsystem intakeSubsystem, IndexSubsystem index
         this.shooterSubsystem = shooterSubsystem;
         this.shooterVisionSubsystem = shooterVisionSubsystem;
         addRequirements(intakeSubsystem, indexSubsystem, shooterSubsystem, shooterVisionSubsystem);
+
     }
 
     @Override
     public void execute() {
 
-        // System.out.println("I AM RUNNING");
-
         boolean ingestSensor = indexSubsystem.ingestSensorHasBallIn();
         boolean feederSensor = indexSubsystem.feederSensorHasBallIn();
         boolean ingestColor = indexSubsystem.ingestHasCorrectCargo() && ingestSensor;
         boolean feederColor = indexSubsystem.feederHasCorrectCargo() && feederSensor;
 
         for (Bitmap value : Bitmap.values()) {
-
             if (value.equals(ingestSensor, feederSensor, ingestColor, feederColor)) {
-                intakeSubsystem.setSpeed(value.intakeMotorSpeed);
-                indexSubsystem.setSpeed(value.ingestMotorSpeed, value.feederMotorSpeed);
-
-                double yaw = shooterVisionSubsystem.getDistance() + shooterSubsystem.getTurretAngleBias();
-                shooterSubsystem.updateTurretAngle(yaw);
-
-                if (value.shooterMisfire) {
-                    shooterSubsystem.setHoodAngle(0);
-                    shooterSubsystem.setFlywheelVelocity(MISFIRE_VELOCITY);
-
-                } else {
-                    double distance = shooterVisionSubsystem.getDistance() + shooterSubsystem.getDistanceBias();
-                    ShooterDataDistancePoint shooterData = ShooterConstants.dataPoints.getInterpolated(distance);
-                    shooterSubsystem.setHoodAngle(shooterData.getAngle());
-                    shooterSubsystem.setFlywheelRPM(shooterData.getRPM());
-                }
+                currentState = value;
+                break;
             }
         }
+
+        indexSubsystem.setBitmapState(currentState);
```

Comment:
If we are worried about overloading the CAN bus, we can consider comparing current state with previous state and skipping the set commands if no change detected.  But, this might be better in a separate commit.